code/game/objects/items.dm

```diff
@@ -699,10 +744,36 @@
 	return 0
 
 //Called when the item blocks an attack. Return 1 to stop the hit, return 0 to let the hit go through
-/obj/item/proc/on_block(damage, attack_text = "the attack")
+/obj/item/proc/on_block(damage, attack_text = "the attack", atom/target, block_sound = null)
 	if(ismob(loc))
-		if(prob(50 - round(damage / 3)))
-			visible_message("<span class='danger'>[loc] blocks [attack_text] with \the [src]!</span>")
+		var/base_block_chance = max(50 - round(damage/3), 15) //Base chance ranges from 50% to 15%. The higher the damage, the lower the chance.
+
+		if(target && (get_turf(target) != get_turf(loc))) //If target isn't standing on the same turf as the attacker
+
+			var/angle = 180 + dir2angle(get_dir(target, loc)) - dir2angle(loc.dir) //This ranges from -180 to 180. 0 means the attacker is in front of the mob we're protecting. 90 means the attacker is attacking fron the side, 180 means the attacker is hitting from behind, ...
+			if(angle > 180) angle-=360
+
+			switch(abs(angle))
+				if(0) //Attack from front - +30% chance
+					base_block_chance += 30
+				if(45) //From diagonal - +10%
+					base_block_chance += 10
+				//if(90) //From side - nothing
+				if(135) //From behind, diagonal - -10%
+					base_block_chance -= 10
+				if(180) //From behind - -30%
+					base_block_chance -= 30
+
+			base_block_chance = Clamp(base_block_chance, 0, 80) //If you're facing away from the attack, the chance to block it may become 0%.
+
+		if(properties["blocking"])
```

Comment:
Yeah, so if you want that on an item level then I suggest adding a GetProperty(string) proc for this.
Also make that a precompile def, you shouldn't have strings if they need to be repeated consistently.